android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -336,9 +345,26 @@ public void savedBandwidth(long savings) {
         }
     }
 
+    @Override
+    public void onConnectionError(MojoException e) {
+        mCookieListOptInPageAndroidHandler = null;
+        initCookieListOptInPageAndroidHandler();
+    }
+
+    private void initCookieListOptInPageAndroidHandler() {
+        if (mCookieListOptInPageAndroidHandler != null) {
+            return;
+        }
+
+        mCookieListOptInPageAndroidHandler =
```

Comment:
we should call `mCookieListOptInPageAndroidHandler.close` when we are done, there are possible leaks otherwise, there are warning messages about that in logcat. A good place for that is
```
@Override
    void destroy() {
```